### PR TITLE
MBS-11235: Convert Edit Instrument edit to React 

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-11235

Only significant change is using IntentionallyRawIcon for name and description.

Also fixed a couple errors in EditEvent while at it.

### CODE REVIEW

**Comment by reosarevok** [comment]
@brainzbot, please retest this.

**Review by mwiencek — APPROVED** [review]

File: root/edit/details/EditInstrument.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            : null}
+          oldContent={type.old
+            ? lp_attributes(type.old.name, 'instrument_type')
+            : null}
+        />
+      ) : null}
+      {description ? (
+        <WordDiff
+          extraNew={rawIconSection}
+          extraOld={rawIconSection}
```
Comment (yvanzo) [inline root/edit/details/EditInstrument.js:83]: Do not display the icon for empty values:

```suggestion
          extraNew={nonEmpty(newDescription) ? rawIconSection : null}
          extraOld={nonEmpty(oldDescription) ? rawIconSection : null}
```

Copied from https://github.com/metabrainz/musicbrainz-server/commit/65090b8089f4f6b65717a0fc54c465aa3aeb3b83#diff-8acf88c2a9626fe10451af2e3c563047c241ad9937c38293c36b0ccd70cdcf7eR77-R78

File: root/edit/details/EditInstrument.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  return (
+    <table className="details edit-event">
+      <tr>
+        <th>{addColonText(l('Instrument'))}</th>
+        <td colSpan="2"><EntityLink entity={display.instrument} /></td>
+      </tr>
+      {name ? (
+        <WordDiff
+          extraNew={rawIconSection}
+          extraOld={rawIconSection}
```
Comment (yvanzo) [inline root/edit/details/EditInstrument.js:56]: Name cannot be empty so this is fine.

**Review by yvanzo — CHANGES_REQUESTED** [review]
Works nicely, just one small glitch with icon for empty values.

File: root/edit/details/EditInstrument.js (hunk 1 above)
Comment (reosarevok) [reply root/edit/details/EditInstrument.js]: Good catch!

**Review by yvanzo — APPROVED** [review]

**Pull request merged**